tutorials/optimization/using_servers.rst

```diff
@@ -0,0 +1,189 @@
+
+Servers
+-------
+
+One of the most interesting design decisions for Godot, is the fact that the whole scene system is
+*optional*. While it is not currently possible to compile it out, it can be completely bypassed.
+
+At the core, Godot uses the concept of Servers. They are very low level APIs to control
+rendering, physics, sound, etc. The scene system is built on top of them and uses them directly.
+The most common servers are:
+
+* :ref:`VisualServer <class_VisualServer>`: handles everything related to graphics.
+* :ref:`PhysicsServer <class_PhysicsServer>`: handles everything related to 3D physics.
+* :ref:`Physics2DServer <class_Physics2DServer>`: handles everything related to 2D physics.
+* :ref:`AudioServer <class_AudioServer>`: handles everything related to audio.
+
+Just explore their APIs and you will realize that the all functions provided are low-level
+implementations of everything Godot allows you to do.
+
+RIDs
+----
+
+The key to using servers is understanding Resource ID (:ref:`RID <class_RID>`) objects. These are opaque
+handles to the sever implementation. They are allocated and freed manually. Almost every
+function in the servers requires RIDs to access the actual resource.
+
+Most Godot nodes and resources contain these RIDs from the servers internally, and they can
+be obtained with different functions. In fact, anything that inherits :ref:`Resource <class_Resource>`
+can be directly casted to an RID (not all resources contain an RID, though, in such cases
+the RID will be empty). In fact, resources can be passed to server APIs as RIDs. Just make
+sure to keep references to the resources ouside the server, because if the resource is erased,
+the internal RID is erased too.
+
+For nodes, there are many functions available:
+
+* For CanvasItem, the :ref:`CanvasItem.get_canvas_item() <class_CanvasItem_method_get_canvas_item>`
+  method will return the canvas item RID in the server.
+* For CanvasLayer, the :ref:`CanvasLayer.get_canvas() <class_CanvasLayer_method_get_canvas>`
+  method will return the canvas RID in the server.
+* For Viewport, the :ref:`Viewport.get_viewport_rid() <class_Viewport_method_get_viewport_rid>`
+  method will return the viewport RID in the server.
+* For 3D, the :ref:`World <class_World>` resource (obtainable in the :ref:`Viewport <class_Viewport>`
+  and :ref:`Spatial <class_Spatial>` nodes)
+  contains functions to get the *VisualServer Scenario*, and the *PhysicsServer Space*. This
+  allows creating 3D objects directly with the server API and using them.
+* For 2D, the :ref:`World2D <class_World2D>` resource (obtainable in the :ref:`Viewport <class_Viewport>`
+  and :ref:`CanvasItem <class_CanvasItem>` nodes)
+  contains functions to get the *VisualServer Canvas*, and the *Physics2DServer Space*. This
+  allows creating 2D objects directly with the server API and using them.
+* The :ref:`VisualInstance<class_VisualInstance>` class, allows getting the scenario *instance* and
```

Comment:
Those two methods were added in the master branch, but I cherrypicked them in 3.1.1-stable, so they should work fine in the current stable.

Might be worth adding a note that they're only in 3.1.1+ though.